src/components/controls/atoms/InstagramPostEmbed.js

```diff
@@ -0,0 +1,295 @@
+import React, { useEffect } from "react";
+
+/**
+ * ! If instagram changes how their embeds works in the future, this component might break at some point.
+ * Meta hides their `oembed_read` API behind very restrictive API scopes that require a Graph API application and a manual review by Meta.
+ * Since we can't do that, this component is a manually derived copy of the "official" instagram embed returned when clicking `Embed` in instagram's UI.
+ * The HTML code is mostly there to show a matching loading preview.
+ * Steps taken to derive React component:
+ *  1. Create a manual embed via the Instagram UI and copy the HTML code
+ *  2. Transform html => jsx
+ *  3. Remove caption and author paragraphs
+ *  4. Replace script loading mechanism
+ *  5. Replace urls
+ */
+function InstagramPostEmbed({ src }) {
+  useEffect(() => {
+    if (window.instgrm) {
+      /*
+       * this method swaps instagram embeds in the DOM with the posts.
+       * since embeds can enter the DOM after the SDK has loaded for the first time, we need to call this manually.
+       */
+      window.instgrm?.Embeds?.process();
+    } else {
+      injectScript("//www.instagram.com/embed.js");
```

Comment:
I wonder what are the implications of injecting scripts from ad platforms or even just hotlinking images/videos hosted there...

I'd imagine that by doing this we are effectively sharing data with Meta Inc about user's interest in these pictures/videos. We should be asking the consent to that.

One easy solution for this problem is to not show embeds without additional confirmation.

- show the info that we have without making any third-party calls
- show links to the third-party content
- load the embed when user indicates that they want to see it

This could be as light as showing the link (which can be opened in a new tab, if they just want to go to the site), but loading the embed when that link is clicked instead of opening the site. The site can still be opened from the embed.

This is hardly a legal-proof solution, but one could argue that clicking an "Instagram" link is an indicator of consent to interact with Meta.

Reply:
This is a valid concern both from a user's perspective as well as from a legal one (GDPR).

I think it would make sense to open an issue for it here or even in the upstream repo to discuss details. It's a bit out of scope for this PR though.

One nice thing about this specific embed is that it's a perfectly styled placeholder link to the content until the insta script loads already.